live-pixel/validator.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * validation functions to verify livePixel json from athena
+ */
+
+class Error {
+    text
+    options
+
+    constructor(text) {
+        this.text = text
+        this.options = {}
+    }
+
+    buildMessage() {
+        let optionsAsString = this.options.join(', ')
+        return this.text + ' ' + optionsAsString
+    }
+}
+
+const urlToJson = require('./models/json-to-query')
+
+function invoke(livePixelUrl, livePixelDBRow) {
+    return validateAutogeneratedValues(livePixelDBRow)
+}
+
+function validateAutogeneratedValues(livePixelDBRow) {
+    let error = new Error('[LivePixel Athena Row] The following fields must have values: ')
+    error.options['keys'] = []
+
+    Object.keys(urlToJson.autogeneratedValues).forEach(key => {
+        let currentValue = livePixelDBRow[key.toLowerCase()]
+        if (!currentValue) {
+            error.options['keys'].push(key)
+        }
+    })
+    if (error.options) {
+        error.options['payload'] = livePixelDBRow
+        return error
+    } else {
+        return true
+    }
+}
+
+function validateDetails(livePixelUrl, livePixelDBRow) {
+    let error = new Error('[LivePixel Athena Row] The following values are not matching')
+    error.options['payload'] = []
+
+    let dbRowDetails = formatDetails(livePixelDBRow)
+    Object.entries(urlToJson.details).forEach(entry => {
+        const [dbKey, queryKey] = entry
+
+        let valueFromDBRow = dbRowDetails[dbKey.toLowerCase()]
+        let valueFromQuery = livePixelUrl.params[queryKey]
+        // console.log('COMPARING ....', valueFromDBRow,'--------', valueFromQuery)
+
+        if (valueFromQuery !== valueFromDBRow) {
+            let current = []
+            current['queryKey'] = valueFromQuery
+            current['dbKey'] = valueFromDBRow
+            error.options['payload'].push(current)
+        }
+    })
+    if (error.options) {
+        return error
+    } else {
+        return true
+    }
```

Comment:
ternary might be handy here
```suggestion
    return error.options ? error : true
```